vault/auth.go

```diff
@@ -817,6 +821,15 @@ func (c *Core) setupCredentials(ctx context.Context) error {
 			// don't set the running version to a builtin if it is running as an external plugin
 			if externaler, ok := backend.(logical.Externaler); !ok || !externaler.IsExternal() {
 				entry.RunningVersion = versions.GetBuiltinVersion(consts.PluginTypeCredential, entry.Type)
+
+				// Shutdown or skip on initial mount, depending on whether or not this is a major/minor upgrade.
+				isNonPatchUpdate := isMajorOrMinorUpgrade(c.currentVaultVersion.Version, entry.LastMounted)
+				if _, err := c.handleDeprecatedMountEntry(ctx, entry, consts.PluginTypeCredential, isNonPatchUpdate); err != nil {
+					backend.Cleanup(ctx)
+					backend = nil
+					c.logger.Error("skipping deprecated credential entry", "path", entry.Path)
```

Comment:
Double logging the error does seem awkward. I think normally you'd only log _and_ return an error value if the log and the return value are going to different consumers, e.g. in an HTTP handler where the error goes to the client, but for some errors you also want to have visibility from the server logs. But for `handleDeprecatedMountEntry`, its return value is always processed by the caller before generating a client error, so for my 2c the logging seems like it would be better handled outside of `handleDeprecatedMountEntry`.